### PR TITLE
[SPARK-50468][BUILD] Upgrade Guava to 33.3.1-jre

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr aims to upgrade guava from 33.2.1-jre to 33.3.1-jre

### Why are the changes needed?
The full release notes as follows:
- https://github.com/google/guava/releases/tag/v33.3.0
- https://github.com/google/guava/releases/tag/v33.3.1

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GitHub Actions

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

**Comment by LuciferYang** [comment]
All Passed

File: project/SparkBuild.scala [hunk 1]
```diff
@@ -1057,7 +1057,7 @@ object KubernetesIntegrationTests {
  * Overrides to work around sbt's dependency resolution being different from Maven's.
  */
 object DependencyOverrides {
-  lazy val guavaVersion = sys.props.get("guava.version").getOrElse("33.1.0-jre")
```
Comment (dongjoon-hyun) [inline project/SparkBuild.scala:1060]: Oh. Thank you for catching this, @LuciferYang .

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by LuciferYang** [comment]
Thanks @dongjoon-hyun

**Pull request closed without merging**